static/app/views/explore/hooks/useTimeseries.tsx

```diff
@@ -0,0 +1,137 @@
+import type {DiscoverDatasets} from 'sentry/utils/discover/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+
+/**
+ * Row represents a point in a timeseries - plotting (timestamp, value) with a line between them would represent a line chart.
+ */
+export interface Row {
+  /** timestamp: The time (in seconds) of this point in the time series */
+  timestamp: number;
+
+  /** value: The float value of this point in the time series (i.e., how high it is in the graph) */
+  value: number;
+
+  // comparisonValue: For alerts - we run two timeseries, where one is some delta
+  // in the past (say, 5 minutes before the 'real timeseries') - this value is the analogue of 'value'
+  // for the second timeseries in the past.
+  // Alerts use this to calculate a relative change in timeseries with ((value - comparisonValue) / comparisonValue) * 100
+  comparisonValue?: number;
+
+  /** confidence: For sampled datasets - whether there is enough data for the timeseries to be accurate. */
+  confidence?: 'low' | 'high';
+
+  /** sampleCount: For sampled datasets - how many samples were used to calculate this datapoint. */
+  sampleCount?: number;
+
+  /** sampleRate: For sampled datasets - 0.1 represents 9/10 events being deleted due to sampling */
+  sampleRate?: number;
+}
+
+/**
+ * SeriesMeta represents metadata about a single timeseries.
+ *
+ * For example, it might say what type of data is in this timeseries.
+ */
+export interface SeriesMeta {
+  /** valueType: The type of the unit, e.g., 'duration' or 'size' */
+  valueType: string;
```

Comment:
This should be a string union of the known types, you can copy `AttributeValueType` from `TimeSeries`, or import it